### PR TITLE
Create player class

### LINKED ISSUE #10: Create Player class
- Player class is responsible for interacting with their Board and this is a Parent class for HumanPlayer and ComputerPlayer

Interaction pattern
```rb
Player::NO_METHOD_MESSAGE
#=> "Needs to be Implemented"

player = Player.new(board) 
player.place_ships
player.take_turn
player.render_board # returns the board

```
All the methods above should be declared as abstract
```rb
def take_turn
   fail NotImplementedError, "A player needs to be able to take a turn"
end
```

### PR DESCRIPTION
**Pull request by redferret**
closes #10

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Player
```
Comment (markcyen) [inline lib/player.rb:1]: Do we need a `require './lib/board`?

File: spec/player_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+require './lib/player'
+require 'rspec'
+
+describe Player do
+  context '#initialize' do
+    it 'exists' do
+      board = double('board')
```
Comment (markcyen) [inline spec/player_spec.rb:7]: Oh `double` is new to me. Would like to learn about this.

File: spec/player_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      expect(player).to be_instance_of Player
+    end
+  end
+
+  context '#render_board' do
+    it 'has a board and calls render on it' do
+      test_string = "  1 2 3 4 \nA . . . . \nB . . . . \nC . . . . \nD . . . . \n"
+
+      board = double('board')
+      allow(board).to receive(:render).and_return(test_string)
```
Comment (markcyen) [inline spec/player_spec.rb:18]: this one is new to me too. so english-y!

**Review by markcyen — APPROVED** [review]
Great! Just a few comments.

File: lib/player.rb (hunk 1 above)
Comment (redferret) [reply lib/player.rb:1]: yes, very good point.

File: spec/player_spec.rb (hunk 2 above)
Comment (redferret) [reply spec/player_spec.rb:7]: mocking a board

**Pull request merged**